keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md

```diff
@@ -254,14 +254,24 @@ the number if needed, but upgrades aren't that frequent and few resources are
 changed on releases, so these requests should not be common. We will count them
 with a metric.
 
+There could be a large volume of requests for a specific resource which might result in the identified apiserver to be unable to serve the proxied requests. Although this scenario should not occur too frequently, since resource types that have large request traffic should not undergo upgrades too often.
+
 TODO: security / cert stuff.
 
 ## Design Details
 
-TODO: specific API change (x2)
+Specific API change(s)
+
+1. In [APIServiceSpec](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/types.go#L50)
+   1. Add a list of serviceableby (alternate API server ids ?) to be referenced to identify which apiserver can handle the incoming request.
+   2. Concern: Should the list of serviceableby servers (mutable by nature) go inside the spec - which should rather be kept immutable?
+2. In [ServerStorageVersion](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/apiserverinternal/types.go#L62):
+   1. Add a list of resource ids that an apiserver can serve
+   2. Add the port of the apiserver instance, to be used in case a request needs to be proxied
+
+To determine if a request is for a resource that should be proxied, the handler should match the groupversion for the local apiserver and that of the request. If they match, the handler should determine if the resource for the request is known by the local apiserver. To do so, the handler can check the StorageVersion API to get a list of known resources for the local apiserver
```

Comment:
I think this and the next paragraph would be better if you wrote pseudocode or a procedural list; as written, it sounds like we'd be reading storage version api twice. And we should address concerns like latency (so we should cache) and accuracy (so we should bound how out of date the cache can be, maybe).